src/generator/type_inference_visitor.ts

```diff
@@ -198,23 +204,31 @@ class TypeInferenceVisitor extends Visitor {
         });
     }
 
-    private updateVariableType(identifier: Identifier, rightSideType: WebAssemblyType) {
-        const variableName = identifier.name;
+    private checkIfTypeIsUnchanged(identifier: Identifier, type: WebAssemblyType) {
+        const name = identifier.name;
 
-        if (this.variableTypes.has(variableName)) {
-            const currentValue = this.variableTypes.get(variableName);
+        if (this.variableTypes.has(name)) {
+            const currentValue = this.variableTypes.get(name);
 
-            if (currentValue !== rightSideType) {
+            if (currentValue !== type) {
                 if (currentValue === undefined) {
-                    throw new Error(`Tried to change the value type of ${variableName}
-                    from undefined to ${WebAssemblyType[rightSideType]}`);
+                    throw new Error(`Tried to change the value type of ${name}
+                    from undefined to ${WebAssemblyType[type]}`);
                 } else {
-                    throw new Error(`Tried to change the value type of ${variableName}
-                    from ${WebAssemblyType[currentValue]} to ${WebAssemblyType[rightSideType]}`);
+                    throw new Error(`Tried to change the value type of ${name}
+                    from ${WebAssemblyType[currentValue]} to ${WebAssemblyType[type]}`);
                 }
             }
-        } else {
-            this.variableTypes.set(variableName, rightSideType);
+        }
+    }
+
+    private updateVariableType(identifier: Identifier, type: WebAssemblyType) {
```

Comment:
Maybe change the name of the method to `updateIdentifierType` and rename `variableTypes` to `identifierTypes`.